src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,39 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int NUM_FIGURE_TYPES = 5;
+    private static final double SIDE_LIMIT = 10.0;
+    private static final int DEFAULT_RADIUS = 10;
+    private static final String DEFAULT_COLOR = "white";
+
+    private final Random random = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int randomType = random.nextInt(NUM_FIGURE_TYPES);
+        String randomColor = new ColorSupplier().getRandomColor();
+        switch (randomType) {
+            case 0:
+                return new Square(getRandomSide(), randomColor);
+            case 1:
+                return new Rectangle(getRandomSide(), getRandomSide(), randomColor);
+            case 2:
+                return new RightTriangle(getRandomSide() + 1, getRandomSide() + 1, randomColor);

```

Comment:
why do you add + 1 only here?

Reply:
Since it's a triangle, and the condition doesn't specify whether it's right or not, it seemed to me that this was the right solution.